bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/MPSBulkSynchronizer.kt

```diff
@@ -94,56 +123,114 @@ object MPSBulkSynchronizer {
+    ) {
+        val access = repository.modelAccess
+        ThreadUtils.runInUIThreadAndWait {
+            access.executeCommand {
                 println("Importing modules...")
-                // `modulesToImport` lazily produces modules to import
-                // so that loaded model data can be garbage collected.
-                val modulesToImport = includedModules.asSequence().flatMapIndexed { index, module ->
-                    println("Importing module ${index + 1} of $numIncludedModules: '${module.moduleName}'")
-                    val fileName = inputPath + File.separator + module.moduleName + ".json"
-                    val moduleFile = File(fileName)
-                    if (moduleFile.exists()) {
-                        val expectedData: ModelData = moduleFile.inputStream().use(Json::decodeFromStream)
-                        sequenceOf(ExistingAndExpectedNode(MPSModuleAsNode(module), expectedData))
-                    } else {
-                        println("Skip importing ${module.moduleName}} because $fileName does not exist.")
-                        sequenceOf()
-                    }
+                try {
+                    println("Importing modules...")
+                    // `modulesToImport` lazily produces modules to import
+                    // so that loaded model data can be garbage collected.
+                    val modulesToImport = getModulesToImport()
+                    ModelImporter(rootOfImport, continueOnError).importIntoNodes(modulesToImport)
+                    println("Import finished.")
+                } catch (ex: Exception) {
+                    // Exceptions are only visible in the MPS log file by default
+                    ex.printStackTrace()
                 }
-                ModelImporter(repoAsNode, continueOnError).importIntoNodes(modulesToImport)
                 println("Import finished.")
-            } catch (ex: Exception) {
-                // Exceptions are only visible in the MPS log file by default
-                ex.printStackTrace()
             }
-            println("Import finished.")
         }
 
-        ApplicationManager.getApplication().invokeAndWait {
+        ThreadUtils.runInUIThreadAndWait {
             println("Persisting changes...")
-            access.executeCommandInEDT {
+            access.executeCommand {
                 enableWorkaroundForFilePerRootPersistence(repository)
+                updateUnsetResolveInfo(repository)
                 repository.saveAll()
             }
             println("Changes persisted.")
         }
     }
 
+    /**
+     * Workaround for MPS not being able to set the `resolveInfo` property on a reference.
+     * This is the case when the concept of the target node cannot be loaded/is not valid.
+     * Without this workaround, the `resolve` attribute in serialized references
+     * (e.g. <ref role="3SLt5I" node="3vHUMVfa0RY" resolve="referencedNodeA" />)
+     * will not be set or updated.
```

Comment:
More severe, it would also be removed when changing order.

Reply:
Done. I updated the comment to about removal of the value.